connector/controllers/main.py

```diff
@@ -94,7 +94,8 @@ def retry_postpone(job, message, seconds=None):
                 if err.pgcode not in PG_CONCURRENCY_ERRORS_TO_RETRY:
                     raise
 
-                retry_postpone(job, unicode(err), seconds=PG_RETRY)
+                retry_postpone(job, unicode(err.pgerror, errors='replace'),
```

Comment:
May be we can replace the call to `unicode` with 
```python
from openerp import tools
...
    retry_postpone(job, tools.ustr(err), seconds=PG_RETRY)
...
```
https://github.com/odoo/odoo/blob/8.0/openerp/loglevels.py#L58